builtin/logical/database/backend.go

```diff
@@ -27,6 +29,29 @@ const (
 	minRootCredRollbackAge = 1 * time.Minute
 )
 
+// metrics collection
+var (
+	gaugeSync = sync.Mutex{}
+	gauges    = map[string]*int32{}
+	gaugeKey  = []string{"secrets", "database", "backend", "connections", "count"}
```

Comment:
Would it make sense to have these be part of `databaseBackend` instead of globals? If the keys and/or labels are common across database backend mounts, would we still be able to aggregate them, if that's the concern around having them local to the struct?

Reply:
I thought it would make more sense to have them as globals. Let me see how `armon/go-metrics` handles duplication and aggregation though if we have multiple gauges with the same name.

Reply:
Okay, I dug in a little, and `armon/go-metrics` does not do any aggregation for us, so different backends will report metrics with the same label and could trample the others backends.

The BackendUUID can be used to differentiate, but it tends to generate metrics with long, inscrutable names in `statsd`, so I'm debating truncating it or replacing it with an atomic counter.